File: Algorithms/012.IntegerToRoman.cpp
class Solution {
public:
    string intToRoman(int num) {
        string symbols[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        int values[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        string s;
        int i = 0;
        while (num != 0) {
            int k = num / values[i];
            for (int j = 0; j < k; j++) {
                s += symbols[i];
                num -= values[i];
            }
            i++;
        }
        return s;
    }
};
int main() {
    Solution s;
    cout << s.intToRoman(1000) << endl;
    return 0;
}

File: Algorithms/py/012.IntegerToRoman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
        values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        i = 0
        s = ""
        while num:
            k = int(num / values[i])
            for j in range(k):
                s += symbols[i]
                num -= values[i]
            i += 1
        return s